tool/tsh/tsh.go

```diff
@@ -2096,26 +2125,45 @@ Loop:
 			case types.OpPut:
 				r, ok := event.Resource.(*types.AccessRequestV3)
 				if !ok {
-					return nil, trace.BadParameter("unexpected resource type %T", event.Resource)
+					return trace.BadParameter("unexpected resource type %T", event.Resource)
 				}
 				if r.GetName() != req.GetName() || r.GetState().IsPending() {
 					log.Debugf("Skipping put event id=%s,state=%s.", r.GetName(), r.GetState())
 					continue Loop
 				}
-				return r, nil
+				return onRequestResolution(cf, tc, r)
 			case types.OpDelete:
 				if event.Resource.GetName() != req.GetName() {
 					log.Debugf("Skipping delete event id=%s", event.Resource.GetName())
 					continue Loop
 				}
-				return nil, trace.Errorf("request %s has expired or been deleted...", event.Resource.GetName())
+				return trace.Errorf("request %s has expired or been deleted...", event.Resource.GetName())
 			default:
 				log.Warnf("Skipping unknown event type %s", event.Type)
 			}
 		case <-watcher.Done():
-			return nil, trace.Wrap(watcher.Error())
+			return trace.Wrap(watcher.Error())
+		}
+	}
+}
+
+func onRequestResolution(cf *CLIConf, tc *client.TeleportClient, req types.AccessRequest) error {
+	if !req.GetState().IsApproved() {
+		msg := fmt.Sprintf("request %s has been set to %s", req.GetName(), req.GetState().String())
+		if reason := req.GetResolveReason(); reason != "" {
+			msg = fmt.Sprintf("%s, reason=%q", msg, reason)
 		}
+		return trace.Errorf(msg)
 	}
+
+	msg := "\nApproval received, getting updated certificates...\n\n"
+	if reason := req.GetResolveReason(); reason != "" {
+		msg = fmt.Sprintf("\nApproval received, reason=%q\nGetting updated certificates...\n\n", reason)
+	}
+	fmt.Fprint(os.Stderr, msg)
+
+	err := reissueWithRequests(cf, tc, req.GetName())
```

Comment:
If the (approved) access request is already expired will this return a user-friendly error message?

Reply:
access requests are deleted as soon as they expire, so the user would get the error `no access request matching "3f893b99-8175-4d11-b370-7775fb697d9e"` and `reissueWithRequests` wouldn't be reached.

Reply:
However if you provide an invalid request, like one for another user, you get: 
```
Approval received, getting updated certificates...

ERROR: invalid access request "ba996cb5-ce6e-4e00-8c49-e97890728e24"
```